### PR TITLE
Utility & Test classes are added

### PR DESCRIPTION
**Pull request by nickchamp1984**

### CODE REVIEW

File: src/main/java/Person.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            if (this instanceof Woman) {
+                this.lastName = partner.getLastName();
+            }
+            partner.registerPartnership(this); // Зареєструвати партнерство іншого об'єкта тут
+        }
+    }
+
+    public void deregisterPartnership(boolean revertLastName) {
+        if (this.partner != null) {
+            this.partner = null;
```
Comment (yurii-voronenko) [inline src/main/java/Person.java:60]: Before this line add `this.partner.partner = null;`

File: src/test/java/PersonTests.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+                { 28, false },
+                {78, true},
+                {66, true}
+        };
+    }
+
+
+    @Test(dataProvider = "ageData")
+    public void testIsRetiredForMan(int age, boolean expected) {
+        Person man = new Man("Nick", "Pruchkovskyi", age);
```
Comment (yurii-voronenko) [inline src/test/java/PersonTests.java:19]: For testing retirement use boundary values like: 64, 65, 66 for man

File: src/test/java/PersonTests.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        Assert.assertEquals(man.isRetired(), expected);
+    }
+
+    @Test(dataProvider = "ageData")
+    public void testIsRetiredForWoman(int age, boolean expected) {
+        Person woman = new Woman("Mila", "Kalkutina", age);
```
Comment (yurii-voronenko) [inline src/test/java/PersonTests.java:25]: For testing retirement use boundary values like: 59, 60, 61 for woman

**Review by yurii-voronenko — APPROVED** [review]
Done
